### ISSUE TITLE
Fonts not rendering properly

### ISSUE THREAD

**Issue by drmarconi** [Azd325/gitkraken#45]
When opening an existing repository, the fonts for the directory & file names in the dialog window are not rendering correctly.

![2017-05-31-175512_1092x810_scrot](https://cloud.githubusercontent.com/assets/6322808/26643849/a1547a54-462a-11e7-9ad5-54cad9fb4abc.png)

This also happens for any action that opens the same window (Browse for SSH keys for example).

I'm running GitKraken 2.6 on Manjaro i3:

libcurl-openssl-1.0 (AUR)
gitkraken 2.6.0-2 (AUR)

Thanks

**Comment by Azd325** [comment]
start from the console and check if you have an error message.
Otherwise is looks like you miss some font or configuration of it.

**Comment by drmarconi** [comment]
Starting from console I don't get any errors, or whilst using GitKraken to try and open a repo for example, however once i closed it i got the following:

$ gitkraken
Node started time: 1496258658914
state:  update-not-available
EVENT: Main process loaded at 310 ms
state:  checking-for-update
state:  update-not-available
EVENT: Starting initial render of foreground window at 4034 ms
EVENT: Startup triggers started at 4185 ms
/usr/bin/gitkraken: line 3:  1187 Segmentation fault      (core dumped) LD_PRELOAD=/usr/lib/libcurl-openssl-1.0.so.4.4.0 /opt/gitkraken/gitkraken "$@"
$

Also, which font is it expected to be installed/configured? I assumed it uses whatever GTK fonts I have configured?

**Comment by KillWolfVlad** [comment]
GitKraken use [Open Sans](https://en.wikipedia.org/wiki/Open_Sans) fonts

Try install [ttf-opensans](https://aur.archlinux.org/packages/ttf-opensans/) package.

But, this font is build-in and I don't think, that installation help you.

I think your system configuration is wrong. Try launch [Atom](https://wiki.archlinux.org/index.php/Atom) and look to fonts in Atom and GitKraken.

P.S. Atom and GitKraken based on [electron](https://github.com/electron/electron), maybe you should check electron [issues](https://github.com/electron/electron/issues).

Gitkraken use electron v1.2.8

P.P.S. 

> /usr/bin/gitkraken: line 3: 1187 Segmentation fault (core dumped) LD_PRELOAD=/usr/lib/libcurl-openssl-1.0.so.4.4.0 /opt/gitkraken/gitkraken "$@"

it is not good

**Comment by binliv** [comment]
same issue.

**Comment by KillWolfVlad** [comment]
Same issue on
Arch Linux GNOME 3 / Mate / KDE
Manjaro i3

I think this is because GitKraken use old electron version (1.28)

**Comment by drmarconi** [comment]
Yes this appears to be issue with electron. There is an open issue for the same behaviour:

https://github.com/electron/electron/issues/9616

**Comment by MarcinWieczorek** [comment]
Same on Arch 4.12.0-rc3-mainline x86_64 GNU/Linux with i3

**Comment by KillWolfVlad** [comment]
Just rebuild everything with latest electron version #39 
![image](https://cloud.githubusercontent.com/assets/18448635/26679778/f8831090-46ef-11e7-9a9e-d57eaa4d75dc.png)

**Comment by LC43** [comment]
any update on this?

**Comment by KillWolfVlad** [comment]
@LC43 
Only mine PR #39, but @MarcinWieczorek doesn't like it :smile_cat:

It remains only to wait for the new version and believe that the bug will fix, or use [GitKraken-Pro-AUR](https://github.com/KillWolfVlad/GitKraken-Pro-AUR) now

If you don't want patch GitKraken and unlock Pro features absolutely free, you should comment this line `node ./bin/gitcracken.js patch-directory --asar-directory ${srcdir}/app -v` in PKGBUILD

**Comment by MarcinWieczorek** [comment]
@KillWolfVlad don't make me report you.

**Comment by KillWolfVlad** [comment]
As you want. I just trying to fix bugs... This was my last comment in this repository.

**Comment by MarcinWieczorek** [comment]
Thank you for trying to fix bugs and actually giving us solutions. I don't like that you try to thrust your crack down our throats with every comment

**Comment by MaxSchambach** [comment]
Any news on this?
I've been having the same problem since a couple of weeks an ArchLinux with GNOME.

**Comment by MarcinWieczorek** [comment]
I'd recommend rebuilding everything as in #39 
FOR NOW.
We're waiting helpless because we don't have the source.

**Comment by reatlat** [comment]
@MarcinWieczorek I have a same issue on my laptop

> OS: Arch Linux x86_64
> Kernel Release: 4.11.9-1-ARCH
> DE: GNOME

before I have a same issue with upworkTimeTracker

for quick solve this problem, I just made a symlink

```
sudo ln -s /usr/share/upwork/libpangoft2-1.0.so.0.4000.5 /opt/gitkraken/libpangoft2-1.0.so.0
```

I think we can use a same solution like @llybin & @msheremet
PKGBUILD file from upwork [https://aur.archlinux.org/packages/upwork/]()

```
# Maintainer: Lev Lybin <lev.lybin at gmail dot com>
# Maintainer: Maksym Sheremet <msheremet at sheremets dot com>

pkgname=upwork
pkgver=4.2.153.0
_rawver=${pkgver//./_}
_hashver="tkzkho5lhz15j08q"
pkgrel=3
pkgdesc="Desktop App (aka Team App) Standard version"
arch=('i686' 'x86_64')
url="https://www.upwork.com/downloads?source=Footer"
license=('custom')
conflicts=('upwork-alpha' 'upwork-appimage' 'upwork-beta')
depends=('alsa-lib' 'gconf' 'gtkglext' 'libgcrypt15' 'libxss' 'libxtst' 'nss')
makedepends=('git' 'gtk-doc')  # needed to build pango
install=upwork.install

source=("LICENSE" "git+https://git.gnome.org/browse/pango#commit=6c5d1d35061a91c3c0792f7720da3f8308ebff65")
md5sums=('1a87c2b530067bcf4c632e49c18af484' 'SKIP')
md5sums_i686=('fa3aa945068e49081e1ac2cc7490032b')
md5sums_x86_64=('6c33e3ff5ae168c840f46887d690fd5e')
# how to get links Standard, Beta, Alpha: grep "https://updates-desktopapp.upwork.com/binaries" ~/.Upwork/Upwork/Logs/* | tail -n 4
source_x86_64=(upwork_amd64_${pkgver}.deb::https://updates-desktopapp.upwork.com/binaries/v${_rawver}_${_hashver}/upwork_amd64.deb)
source_i686=(upwork_i386_${pkgver}.deb::https://updates-desktopapp.upwork.com/binaries/v${_rawver}_${_hashver}/upwork_i386.deb)

prepare() {
    cd "${srcdir}"
    tar -zxf "${srcdir}/data.tar.gz"
### BEGIN pango section ###
    cd pango
    NOCONFIGURE=1 ./autogen.sh
### END pango section ###
}

### BEGIN pango section ###
build() {
  cd "${srcdir}/pango"
  ./configure --prefix=/usr
  sed -i -e 's/ -shared / -Wl,-O1,--as-needed\0/g' libtool
  make
}

check() {
  cd "${srcdir}/pango"
  make -k check || :
}
### END pango section ###

package() {
    cd "${srcdir}"
    cp -rp usr "${pkgdir}/usr"

    install -Dm644 "${srcdir}/LICENSE" "${pkgdir}/usr/share/licenses/${pkgname}/LICENSE"
### BEGIN pango section ###
    install -Dm755 "${srcdir}/pango/pango/.libs/libpangoft2-1.0.so.0.4000.5" \
                   "${pkgdir}/usr/share/upwork/libpangoft2-1.0.so.0.4000.5"
    ln -s "libpangoft2-1.0.so.0.4000.5" "${pkgdir}/usr/share/upwork/libpangoft2-1.0.so.0"
### END pango section ###
}
```

**Comment by Daemoen** [comment]
Attempted the symlink solution above via both 64 bit and i686 libpango on Fedora 26 (64 bit).  Neither works.  Version of libpango in F26 is :  /usr/lib64/libpangoft2-1.0.so.0.4000.6

**Comment by reatlat** [comment]
@Daemoen I use pure Arch 64bit
I just try use my system lib `/usr/lib64/libpangoft2-1.0.so.0.4000.6` and for me fonts not rendering properly also.

maybe this issue comes with new version of this lib?

try to use old one `libpangoft2-1.0.so.0.4000.5`

**Comment by Daemoen** [comment]
@reatlat That works.  I had to go back to F25's pango package, but it does indeed do the trick.  I'd see if you can do similar with Arch to get the .5 version.

**Comment by KGCybeX** [comment]
@reatlat works for me too.  

I am  Archlinux x64.

**What I did:**
I visited the pango github repository, downloaded the build on 14 April (4000.5), extracted and ran commands in the extracted folder:
- `./configure` 
- ` sudo make`

Then found the files below and appended `.old` to them.
- `/usr/lib64/libpangoft2-1.0.so.0.4000.7` (or whatever version you have) 
- `/usr/lib64/libpangoft2-1.0.so.0` 
- `/usr/lib64/libpangoft2-1.0.so`

then replaced them with the built libs from the older pango version (that which we just built):
- `/usr/lib64/libpangoft2-1.0.so.0.4000.5`
- `/usr/lib64/libpangoft2-1.0.so.0` 
- `/usr/lib64/libpangoft2-1.0.so`

GitKraken text displayed perfectly.

**Comment by reatlat** [comment]
#48

**Comment by Azd325** [comment]
Fixed in #47

**Comment by jbuisine** [comment]
For my part, problem was fixed by using looking this [link](https://www.reddit.com/r/archlinux/comments/6dodbn/unrecognized_characters_in_gtk_file_chooser/).

I downgrade pango package to 1.40.5-1 and ignore update by adding "IgnorePkg=pango" to /etc/pacman.conf.
